server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -274,7 +278,18 @@ private void executeHealth(
 
         final Predicate<ClusterState> validationPredicate = newState -> validateRequest(request, newState, waitCount);
         if (validationPredicate.test(currentState)) {
-            listener.onResponse(getResponse(request, currentState, waitCount, TimeoutState.OK));
+            ClusterHealthResponse clusterHealthResponse = getResponse(request, currentState, waitCount, TimeoutState.OK);
+            if (request.ensureNodeWeighedIn() && clusterHealthResponse.hasDiscoveredClusterManager()) {
+                DiscoveryNode localNode = clusterService.state().getNodes().getLocalNode();
```

Comment:
Why are we picking up the new state of the cluster and not the one where the observer actually succeeded. IMO, this should be `currentState.getNodes().getLocalNode()`. Because, what if something has changed in the state and entire health action was ran on a previous state. This might give misleading results

Reply:
didn't understand this completely. Do you mean `clusterService.state()` may provide new uncommitted state? If that is case, I understand healthcheck wait for cluster state change events to complete and cluster service also has committed cluster state. Let me know if I am missing anything?